bundler/spec/support/platforms.rb

```diff
@@ -65,12 +65,10 @@ def not_local_tag
     end
 
     def local_ruby_engine
-      ENV["BUNDLER_SPEC_RUBY_ENGINE"] || RUBY_ENGINE
+      RUBY_ENGINE
     end
 
     def local_engine_version
-      return ENV["BUNDLER_SPEC_RUBY_ENGINE_VERSION"] if ENV["BUNDLER_SPEC_RUBY_ENGINE_VERSION"]
```

Comment:
This and the above are already a mechanism to simulate another Ruby engine without changing the RUBY_* constants.
It could also be done more directly by stubbing those methods during the tests needing it.